release_docs/RELEASE.txt

```diff
@@ -431,6 +431,198 @@ New Features
+        If this macro is not defined but H5_HAVE_COMPLEX_NUMBERS is defined, the
+        complex number types supported are the MSVC types.
+
+        - H5_SIZEOF_FLOAT_COMPLEX
+
+        This macro is defined in H5pubconf.h and will have a value corresponding
+        to the size of the native float complex datatype, as computed by sizeof().
+        If C99 complex number support is available, this will be the size of the
+        "float _Complex" type. Otherwise, it will be the size of the "_Fcomplex"
+        type. It will have the value 0 if support for a native float complex datatype
+        is not available.
+
+        - H5_SIZEOF_DOUBLE_COMPLEX
+
+        This macro is defined in H5pubconf.h and will have a value corresponding
+        to the size of the native double complex datatype, as computed by sizeof().
+        If C99 complex number support is available, this will be the size of the
+        "double _Complex" type. Otherwise, it will be the size of the "_Dcomplex"
+        type. It will have the value 0 if support for a native double complex datatype
+        is not available.
+
+        - H5_SIZEOF_LONG_DOUBLE_COMPLEX
+
+        This macro is defined in H5pubconf.h and will have a value corresponding
+        to the size of the native long double complex datatype, as computed by sizeof().
+        If C99 complex number support is available, this will be the size of the
+        "long double _Complex" type. Otherwise, it will be the size of the "_Lcomplex"
+        type. It will have the value 0 if support for a native long double complex
+        datatype is not available.
+
+        - H5T_NATIVE_FLOAT_COMPLEX
+
+        This macro maps to the ID of an HDF5 datatype representing the native C
+        float complex datatype (either "float _Complex" or "_Fcomplex") for the
+        platform. If support for a native float complex datatype is not available
+        (H5_HAVE_COMPLEX_NUMBERS is not defined), the macro will map to
+        H5I_INVALID_HID and should not be used.
+
+        - H5T_NATIVE_DOUBLE_COMPLEX
+
+        This macro maps to the ID of an HDF5 datatype representing the native C
+        double complex datatype (either "double _Complex" or "_Dcomplex") for the
+        platform. If support for a native double complex datatype is not available
+        (H5_HAVE_COMPLEX_NUMBERS is not defined), the macro will map to
+        H5I_INVALID_HID and should not be used.
+
+        - H5T_NATIVE_LDOUBLE_COMPLEX
+
+        This macro maps to the ID of an HDF5 datatype representing the native C
+        long double complex datatype (either "long double _Complex" or "_Lcomplex")
+        for the platform. If support for a native long double complex datatype is
+        not available (H5_HAVE_COMPLEX_NUMBERS is not defined), the macro will map
+        to H5I_INVALID_HID and should not be used.
+
+        - H5T_CPLX_IEEE_F16LE / H5T_CPLX_IEEE_F16BE
```

Comment:
The current datatype docs say that datatype names usually follow a "H5T_arch_base" convention, but that made these names a bit long so I chose this convention for simplicity. Names can be changed if there is a better alternative.

Reply:
Request.  Use the full word "COMPLEX", not abbreviation, in type names, to improve readability etc.  An extra three characters is not important.  **H5T_COMPLEX_IEEE_F16LE**, etc.

Reply:
Sure, no issue with renaming these if "H5T_COMPLEX_IEEE_F16LE" is better for readability.

Reply:
Agree - "COMPLEX" is more clear.